### PR TITLE
Feature/spline 281 split gateway apis

### PR DESCRIPTION
**Pull request by wajda**

### CODE REVIEW

File: rest-api-doc-generator/src/main/scala/za/co/absa/spline/swagger/SwaggerDocGenAppContext.scala [hunk 1]
```diff
@@ -19,23 +19,22 @@ package za.co.absa.spline.swagger
...
   override def createBeanFactory: DefaultListableBeanFactory = new MockingBeanFactory(getInternalParentBeanFactory)
 
   register(
-    classOf[WebMvcConfigurationSupport],
     classOf[SwaggerConfig],
     classOf[JacksonConfig],
 
-    classOf[ConsumerRESTConfig],
-    classOf[ProducerRESTConfig])
+    // todo: replace the following compile time dependencies with the runtime parameters. See https://github.com/AbsaOSS/spline/issues/255
```
[the remark this answers is not in the document]
Comment (wajda) [reply rest-api-doc-generator/src/main/scala/za/co/absa/spline/swagger/SwaggerDocGenAppContext.scala]: done in the next PR

File: producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/ProducerRESTConfig.scala [hunk 2]
```diff
@@ -16,12 +16,25 @@
...
+@EnableWebMvc
 @Configuration
+@Import(Array(classOf[JacksonConfig]))
 @ComponentScan(basePackageClasses = Array(
   classOf[controller._package]
 ))
-class ProducerRESTConfig extends WebMvcConfigurer {}
+class ProducerRESTConfig extends WebMvcConfigurer {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
```
[the remark this answers is not in the document]
Comment (wajda) [reply producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/ProducerRESTConfig.scala]: By CQC, for better visibility.

File: consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/ConsumerRESTConfig.scala [hunk 3]
```diff
@@ -20,27 +20,31 @@ import java.util
...
 
+@EnableWebMvc
 @Configuration
+@Import(Array(classOf[JacksonConfig]))
 @ComponentScan(basePackageClasses = Array(
   classOf[controller._package]
 ))
 class ConsumerRESTConfig extends WebMvcConfigurer {
 
+  import scala.concurrent.ExecutionContext.Implicits.global
```
[the remark this answers is not in the document]
Comment (wajda) [reply consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/ConsumerRESTConfig.scala]: By CQC, for better visibility.

File: commons/src/main/scala/za/co/absa/spline/common/webmvc/EstimableFutureReturnValueHandlerSupport.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.context.request.WebRequest
+import za.co.absa.spline.common.future.EstimableFuture
+
+trait EstimableFutureReturnValueHandlerSupport extends ScalaFutureMethodReturnValueHandler {
+  protected val minEstimatedTimeout: Long
+  protected val durationToleranceFactor: Double
+
+  override type F = EstimableFuture[_]
+
+  override protected def getFutureTimeout(future: F, req: WebRequest): Option[Long] = super
```
[the remark this answers is not in the document]
Comment (wajda) [reply commons/src/main/scala/za/co/absa/spline/common/webmvc/EstimableFutureReturnValueHandlerSupport.scala]: It's not a getter.

File: rest-api-doc-generator/src/main/scala/za/co/absa/spline/swagger/SwaggerDocGenAppContext.scala [hunk 5]
```diff
@@ -19,23 +19,22 @@ package za.co.absa.spline.swagger
...
+    classOf[ConsumerRESTConfig]
+    // classOf[ProducerRESTConfig]
```
Comment (leborgneromain) [inline rest-api-doc-generator/src/main/scala/za/co/absa/spline/swagger/SwaggerDocGenAppContext.scala]: should be removed

Comment (wajda) [reply rest-api-doc-generator/src/main/scala/za/co/absa/spline/swagger/SwaggerDocGenAppContext.scala]: yep, this is already removed and fixed in my next PR

**Pull request merged**